autoload/ctrlp.vim

```diff
@@ -2735,8 +2742,29 @@ fu! ctrlp#init(type, ...)
 	en
 	cal s:BuildPrompt(1)
 	if s:keyloop | cal s:KeyLoop() | en
-	return 1
+	retu 1
+endf
+
+" - Events {{{1
+fu! s:notifyChange()
+	let l:cb = s:getextvar('change')
+	if l:cb != -1
+		cal eval(l:cb)
+	en
+endf
+
+fu! ctrlp#set(lines)
+	let g:ctrlp_lines = a:lines
+endf
+
+fu! ctrlp#get()
+	retu g:ctrlp_lines
```

Comment:
ditto

Reply:
I want to avoid primarily because I'm expecting extensions to use lot of items so avoiding deep copy as much as possible unless is really needed.